moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -532,9 +539,24 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
     {
       RCLCPP_DEBUG_STREAM(logger_, "Joint velocity limit scaling applied by a factor of " << joint_limit_scale);
     }
-
     target_state.velocities *= joint_limit_scale;
 
+    // Scale down the acceleration based on joint acceleration limit if applicable.
+    // TODO move to function: ApplyAccelerationLimit(...)
+    Eigen::VectorXd accel = (target_state.velocities - current_state.velocities) / (servo_params_.publish_period);
+    double min_scaling_factor = 1.0;
+    for (double val : accel)
```

Comment:
Const. Also, the names `val` and `accel` aren't descriptive enough -- maybe call the vector `target_accelerations` and then do

```
for (const double accel : target_accelerations)
```